azure/defaults.go

```diff
@@ -143,6 +143,11 @@ func GenerateAvailabilitySetName(clusterName, nodeGroup string) string {
 	return fmt.Sprintf("%s_%s-as", clusterName, nodeGroup)
 }
 
+// WithCount appends the count as suffix to a generated name
+func WithCount(name string, n int) string {
```

Comment:
```suggestion
func WithIndex(name string, n int) string {
```

is this more accurate?

Reply:
Yeah, I think it makes more sense.